src/app/game-state/inventory.service.ts

```diff
@@ -50,7 +50,8 @@ export interface Pill extends Item {
 }
 
 export interface Furniture extends Item {
-  slot: FurniturePosition
+  slot: FurniturePosition,
+  previousId?: string
```

Comment:
Using previous ID instead of some static tier takes care of weird cases like bed of nails. It isn't previous to heated bed, so it will be skipped when progressively upgrading.

Still, any suggestions on a better way to store and handle this are welcome!

Reply:
Also, it being optional allows for items that don't have previous "iterations", like all workshop items.

Reply:
Seems like it should work. I can't think of a better way off the top of my head.